test/admin.test.js

```diff
@@ -0,0 +1,211 @@
+const request = require('supertest'),
+  app = require('../app.js'),
+  validationErrorStatus = 401,
+  correctlyUpdatedStatus = 200,
+  correctlyCreatedStatus = 200,
+  User = require('../app/models').user,
+  { factory } = require('factory-girl'),
+  { hashPassword } = require('../app/helpers/hasher'),
+  firstName = 'fn',
+  lastName = 'ln',
+  correctPassword = 'password',
+  correctEmail = 'email@wolox.com.ar',
+  correctAdminEmail = 'emailAdmin@wolox.com.ar',
+  badToken = 'token';
+
+describe('POST /admin/users', () => {
+  describe('Test invalid password', () => {
+    test('Create user with short password', () =>
+      request(app)
+        .post('/admin/users')
+        .send({
+          first_name: firstName,
+          last_name: lastName,
+          email: correctEmail,
+          password: 'pass',
+          token: badToken
+        })
+        .then(response => {
+          expect(response.status).toBe(validationErrorStatus);
+        }));
+
+    test('Create user with password with invalid characters', () =>
+      request(app)
+        .post('/admin/users')
+        .send({
+          first_name: firstName,
+          last_name: lastName,
+          email: correctEmail,
+          password: 'passwo..rd',
+          token: badToken
+        })
+        .then(response => {
+          expect(response.status).toBe(validationErrorStatus);
+        }));
+  });
+
+  describe('Test invalid email', () => {
+    test('Create user with email with domain different from wolox', () =>
+      request(app)
+        .post('/admin/users')
+        .send({
+          first_name: firstName,
+          last_name: lastName,
+          email: 'email@email.com.ar',
+          password: correctPassword,
+          token: badToken
+        })
+        .then(response => {
+          expect(response.status).toBe(validationErrorStatus);
+        }));
+  });
+
+  describe('Test used mail to change permissions', () => {
+    test('Create not-admin user and change permissions', () => {
+      factory.define('userNotAdmin', User, buildOptions => ({
+        first_name: buildOptions.firstName,
+        last_name: buildOptions.lastName,
+        email: buildOptions.email,
+        password: buildOptions.password,
+        isAdmin: false
+      }));
+
+      factory.define('userAdmin', User, buildOptions => ({
+        first_name: buildOptions.firstName,
+        last_name: buildOptions.lastName,
+        email: buildOptions.email,
+        password: buildOptions.password,
+        isAdmin: true
```

Comment:
We usually have a separate file for defining factories. Its usually `test/support/factories.js`.
Also, you can just do 
```
      factory.define('userAdmin', User, buildOptions => ({
        first_name: chance.name(),
        last_name: chance.last(),
        email: chance.email({domain: "wolox.com"}),
        password: factory.chance('string', { pool: 'abcde' }),
        isAdmin: true
```
To randomly generate the params that are not integral to the factory (Its called 'userAdmin', so that`isAdmin: true` is the only critical thing). Take a look at the [chance](https://chancejs.com/index.html) library that factory-girl uses.

You can still overwrite params as needed in the same way you've been doing when you `.create`.
Keep this in mind to avoid unnecessary setup, like manually setting the first and last name. You are not going to check for then anyways.